### ISSUE TITLE
ModelCheckpoint can't use "val_acc" even if fit() function receives validation_data argument

### ISSUE THREAD

**Issue by haimat** [tensorflow/tensorflow#32239]
**System information**
- OS Platform and Distribution: Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.0.0-rc0-gpu
- Python version: 3.6.8
- CUDA/cuDNN version: 10.0
- GPU model and memory: NVidia Tesla P100 (16 gb)

**Describe the current behavior**
Recently I switched from TF1 wrapped with native Keras to TF2 using the built-in `tf.keras` implementation. In my code I train my model like so:

    checkpoint = ModelCheckpoint(auto_save_path, monitor="val_acc", save_best_only=True)
    callbacks.append(checkpoint)

    keras_model.fit_generator(training_img_generator,
        steps_per_epoch=100, epochs=20, validation_steps=100,
        validation_data=validation_img_generator, callbacks=callbacks)

However, with that code I do get the following warning message:

    WARNING:tensorflow:Can save best model only with val_acc available, skipping.

A full working example can be found as [Github Gist here](https://gist.github.com/haimat/65e9dddc9cd8004ea8870853f84d7b02).

**Describe the expected behavior**
I don't understand this error, because I do provide the `validation_data` argument to the `fit_generator()` function. Also, I used exactly the same code with Keras on top of TF1, which worked fine there. This warning came now after switching to TF2 and built-in Keras API.

**Comment by gowthamkpr** [comment]
@haimat Can you  please provide the github gist of this issue. Thanks!

**Comment by haimat** [comment]
@gowthamkpr I have created a minimal working example as [Github Gist here](https://gist.github.com/haimat/65e9dddc9cd8004ea8870853f84d7b02).

**Comment by yoke2** [comment]
@haimat I'm facing the same issue as you when using ModelCheckpoint in 2.0-rc0. As a work-around, I used tensorflow-gpu==2.0.0-beta1 instead and changed 'val_acc' to 'val_accuracy' and the warning disappeared.

By the way, if you're using CSV logger, please also note that column names 'acc'  and 'val_acc' has been changed to 'accuracy' and 'val_accuracy' respectively.

**Comment by haimat** [comment]
Any chance this will be fixed for the next RC or the final 2.0 release?

**Comment by omalleyt12** [comment]
> is there some kind of workaround

Looking into this issue, in the meantime compiling with "accuracy" and using "val_accuracy" in ModelCheckpoint should work

**Comment by haimat** [comment]
Hey guys, has this been fixed in rc2?

**Comment by omalleyt12** [comment]
@haimat Sorry for the late reply, in TF2.0 it should use whatever you compiled with, i.e. `model.compile(..., metrics=['acc'])` will log as "val_acc", otherwise it will log as "val_accuracy"
